addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200ThingHandler.java

```diff
@@ -0,0 +1,451 @@
+                        continue;
+                    }
+                    /* Check whether the value is a dummy (e.g. not connected sensor) */
+                    val = (BigDecimal) serObj.serviceTreeMap.get(subKey).getValue();
+                    if (val != null) {
+                        if (val.setScale(6, RoundingMode.HALF_UP).equals(maxInt16AsFloat)
+                                || val.setScale(6, RoundingMode.HALF_UP).equals(minInt16AsFloat)
+                                || val.setScale(4, RoundingMode.HALF_UP).equals(maxInt16AsInt)) {
+                            continue;
+                        }
+                    }
+                    /* Check the capabilities of this service */
+                    if (serObj.serviceTreeMap.get(subKey).getValueParameter() != null) {
+                        /* Creating an new channel type with capabilities from service */
+                        // The type is definitely correct here
+                        @SuppressWarnings("unchecked")
+                        List<Object> subValParas = (List<Object>) serObj.serviceTreeMap.get(subKey).getValueParameter();
+                        minVal = (BigDecimal) subValParas.get(0);
+                        maxVal = (BigDecimal) subValParas.get(1);
+                        if (subValParas.size() > 2) {
+                            unitOfMeasure = (String) subValParas.get(2);
+                            if ("C".compareTo(unitOfMeasure) == 0) {
+                                unitOfMeasure = "°C";
+                            }
+                        }
+                        step = BigDecimal.valueOf(0.5);
+                    }
+                    state = new StateDescription(minVal, maxVal, step, "%.1f " + unitOfMeasure, readOnly, null);
+                    newChannel = createChannel(channelTypeUID, channelUID, root, "Number", null, subKey, subKey, true,
+                            false, state, unitOfMeasure);
+                    break;
+
+                case "refEnum":
+                    /* Check whether the sub service should be ignored */
+                    Boolean ignoreIt = false;
+                    for (KM200ThingType tType : KM200ThingType.values()) {
+                        if (tType.getThingTypeUID().equals(thing.getThingTypeUID())) {
+                            for (String ignore : tType.ignoreSubService()) {
+                                if (ignore.equals(subKey)) {
+                                    ignoreIt = true;
+                                }
+                            }
+                        }
+                    }
+                    if (ignoreIt) {
+                        continue;
+                    }
+                    /* Search for new services in sub path */
+                    addChannels(serObj.serviceTreeMap.get(subKey), thing, subChannels, subKey + "_");
+                    break;
+
+                case "errorList":
+                    switch (subKey) {
```

Comment:
Please don't nest switches, it might be a good idea to put the bodies of the switches in their own methods

Reply:
OK, I changed this code. I think it's better to read now.